plugins/client/grpc/client.go

```diff
@@ -100,8 +102,15 @@ func (c *Client) Prepare() error {
 		"client_name": Name,
 	})})
 
+	// server address resolver
+	resolvers.RegisterAllGrpcResolvers()
+
 	// connect to server
-	client, err := grpc.Dial(c.ServerAddr, *cfg...)
+	target, err := resolvers.BuildTarget(&c.ServerFinderConfig)
```

Comment:
Notice, this target could be changed dynamically. Such as in k8s, a domain name of services could have dynamic list of pods IP eventually. We need to resolve this case.

Reply:
This is one of the key issue, ALS client can't dynamically routine requests to all OAP backends.
We need to get the list periodically, and make sure the list getting updated, meanwhile all data(traces/logs/etc.) gets to be sent through one channel.

Reply:
Yes, For the k8s case, there just save the Kubernetes pod/service config(by label selector), not include real address config. The static address config is a special case because we just need to specify the address list.
We use resolver to update the server address list. Please take a look on: https://github.com/apache/skywalking-satellite/pull/68/files#diff-31cdd706c94e7a14dab4bf57861b31fc5b7e919d946fd480c593f3f18f8aafd4R69-R76